client/me/notification-settings/reader-subscriptions/index.jsx

```diff
@@ -25,270 +25,259 @@ import { recordGoogleEvent } from 'calypso/state/analytics/actions';
+							<option value="text">{ translate( 'Plain Text' ) }</option>
+						</FormSelect>
+					</FormFieldset>
 
-							<FormSelect
-								disabled={ this.props.getDisabledState() }
-								id="subscription_delivery_hour"
-								name="subscription_delivery_hour"
-								onChange={ this.props.updateSetting }
-								onFocus={ this.handleFocusEvent( 'Email Delivery Window Time' ) }
-								value={ this.props.getSetting( 'subscription_delivery_hour' ) }
-							>
-								<option value="0">{ this.getDeliveryHourLabel( 0 ) }</option>
-								<option value="2">{ this.getDeliveryHourLabel( 2 ) }</option>
-								<option value="4">{ this.getDeliveryHourLabel( 4 ) }</option>
-								<option value="6">{ this.getDeliveryHourLabel( 6 ) }</option>
-								<option value="8">{ this.getDeliveryHourLabel( 8 ) }</option>
-								<option value="10">{ this.getDeliveryHourLabel( 10 ) }</option>
-								<option value="12">{ this.getDeliveryHourLabel( 12 ) }</option>
-								<option value="14">{ this.getDeliveryHourLabel( 14 ) }</option>
-								<option value="16">{ this.getDeliveryHourLabel( 16 ) }</option>
-								<option value="18">{ this.getDeliveryHourLabel( 18 ) }</option>
-								<option value="20">{ this.getDeliveryHourLabel( 20 ) }</option>
-								<option value="22">{ this.getDeliveryHourLabel( 22 ) }</option>
-							</FormSelect>
+					<FormFieldset>
+						<FormLabel htmlFor="subscription_delivery_day">
+							{ translate( 'Email delivery window' ) }
+						</FormLabel>
+						<FormSelect
+							disabled={ props.getDisabledState() }
+							className="reader-subscriptions__delivery-window"
+							id="subscription_delivery_day"
+							name="subscription_delivery_day"
+							onChange={ props.updateSetting }
+							onFocus={ handleFocusEvent( 'Email delivery window day' ) }
+							value={ props.getSetting( 'subscription_delivery_day' ) }
+						>
+							<option value="0">{ translate( 'Sunday' ) }</option>
+							<option value="1">{ translate( 'Monday' ) }</option>
+							<option value="2">{ translate( 'Tuesday' ) }</option>
+							<option value="3">{ translate( 'Wednesday' ) }</option>
+							<option value="4">{ translate( 'Thursday' ) }</option>
+							<option value="5">{ translate( 'Friday' ) }</option>
+							<option value="6">{ translate( 'Saturday' ) }</option>
+						</FormSelect>
 
-							<FormSettingExplanation>
-								{ this.props.translate(
-									'When choosing daily or weekly email delivery, which time of day would you prefer?'
-								) }
-							</FormSettingExplanation>
-						</FormFieldset>
+						<FormSelect
+							disabled={ props.getDisabledState() }
+							id="subscription_delivery_hour"
+							name="subscription_delivery_hour"
+							onChange={ props.updateSetting }
+							onFocus={ handleFocusEvent( 'Email Delivery Window Time' ) }
+							value={ props.getSetting( 'subscription_delivery_hour' ) }
+						>
+							{ [ ...Array( 12 ).keys() ].map( ( i ) => (
```

Comment:
This will be needlessly called on every re-render though - array creation, key extraction, looping, option creating. This entire call is a perfect scenario for moving to a pure component that will seldom re-render. If you don't want to further componentize, creating the array still makes sense to be done outside so it happens only once.